habitat/core/env.py

```diff
@@ -165,7 +165,7 @@ def _reset_stats(self) -> None:
         self._elapsed_steps = 0
         self._episode_over = False
 
-    def reset(self) -> Observations:
+    def reset(self, keep_current_episode=False) -> Observations:
```

Comment:
I would advice not to add `keep_current_episode` parameter to `Env.reset` as we don't have enough use-cases and there is another way to get around here using [iterator options in config](https://github.com/facebookresearch/habitat-api/blob/master/habitat/config/default.py#L25):
```
ENVIRONMENT.ITERATOR_OPTIONS.CYCLE = True
ENVIRONMENT.ITERATOR_OPTIONS.NUM_EPISODE_SAMPLE = 1
```.
That will repeat same episode over and over again.